### PR TITLE
fix(select menu): include fieldset around checkbox groups

### LINKED ISSUE #1407: Select menu with checkbox groups should include fieldset
Whenever a group of checkboxes are present, they should be grouped within a `<fieldset>`.

The example labeled "Select - Checkbox (Expanded and Selected with Groups)" includes two checkbox groups, but no `<fieldset>`.

- Remove `pf-c-form__fieldset` from `.pf-c-form__group `
- Include `<fieldset class="pf-c-form__group" aria-labelledby="[id of group label]">` after `.pf-c-select__menu-group-title`
- Add `id` and `aria-hidden="true"` to `.pf-c-select__menu-group-title`

```
<div class="pf-c-select__menu-group">
  <div aria-hidden="true" id="group-status" class="pf-c-select__menu-group-title">Status</div>
    <fieldset class="pf-c-form__fieldset" aria-labelledby="group-status">        
      <label class="pf-c-check pf-c-select__menu-item" for="select-checkbox-expanded-selected-running">
```

### PR DESCRIPTION
**Pull request by christiemolloy**
Fix #1407

### CODE REVIEW

File: src/patternfly/components/Select/select-checkbox.hbs [hunk 1]
```diff
@@ -3,28 +3,28 @@
...
-      {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-degraded" name="' id '-degraded"')}}{{/check-input}}
-      {{#> check-label check-label--type="span"}}Degraded{{/check-label}}
-    {{/check}}
-    {{#> check check--type="label" check--attribute=(concat 'for="' id '-needsMaintenance"') check--modifier="pf-c-select__menu-item"}}
-      {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-needsMaintenance" name="' id '-needsMaintenance"')}}{{/check-input}}
-      {{#> check-label check-label--type="span"}}Needs Maintenance{{/check-label}}
-    {{/check}}
-  </fieldset>
-  </form>
+  {{#> form novalidate}}
```
Comment (mcoker) [inline src/patternfly/components/Select/select-checkbox.hbs]: Can you remove `novalidate`? It comes with `form.hbs`

File: src/patternfly/components/Select/select-checkbox-groups.hbs [hunk 2]
```diff
@@ -4,44 +4,48 @@
...
-        {{/check}}
-        {{#> check  check--type="label" check--attribute=(concat 'for="' id '-degraded"') check--modifier="pf-c-select__menu-item"}}
-          {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-degraded" name="degraded"')}}{{/check-input}}
-          {{#> check-label check-label--type="span"}}Degraded{{/check-label}}
-        {{/check}}
-        {{#> check  check--type="label" check--attribute=(concat 'for="' id '-needsMaintenance"') check--modifier="pf-c-select__menu-item"}}
-          {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-needsMaintenance" name="needsMaintenance"')}}{{/check-input}}
-          {{#> check-label check-label--type="span"}}Needs Maintenance{{/check-label}}
-        {{/check}}
+        <div aria-hidden="true" id="group-status" class="pf-c-select__menu-group-title">Status</div>
```
Comment (mcoker) [inline src/patternfly/components/Select/select-checkbox-groups.hbs]: can you make `id="group-status"` unique so it doesn't conflict with the other example? Maybe use `{{id}}` to generate the se ID's and the `aria-labelledby` values?

File: src/patternfly/components/Select/select-checkbox-groups.hbs [hunk 3]
```diff
@@ -4,44 +4,48 @@
...
-        {{/check}}
-        {{#> check  check--type="label" check--attribute=(concat 'for="' id '-samsung"') check--modifier="pf-c-select__menu-item pf-m-disabled"}}
-          {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-samsung" name="samsung" disabled')}}{{/check-input}}
-          {{#> check-label check-label--type="span" check-label--modifier="pf-m-disabled"}}Samsung{{/check-label}}
-        {{/check}}
-        {{#> check check--type="label" check--attribute=(concat 'for="' id '-hp"') check--modifier="pf-c-select__menu-item"}}
-          {{#> check-input check-input--attribute=(concat 'type="checkbox" id="' id '-hp" name="hp"')}}{{/check-input}}
-          {{#> check-label check-label--type="span"}}Hewlett-Packard{{/check-label}}
-        {{/check}}
+        <div aria-hidden="true" id="group-vendor" class="pf-c-select__menu-group-title">Vendor</div>
```
Comment (mcoker) [inline src/patternfly/components/Select/select-checkbox-groups.hbs]: same as above

File: src/patternfly/components/Select/select-checkbox-groups.hbs (hunk 2 above)
Comment (christiemolloy) [reply src/patternfly/components/Select/select-checkbox-groups.hbs]: Yes!

**Pull request merged**